src/python/CMSMonitoring/StompAMQ.py

```diff
@@ -19,19 +19,41 @@
 
 # global object which holds CMS Monitoring schemas
 _schemas = Schemas(update=3600, jsonschemas=False)
+_local_schemas = None
+
+def validate(doc, schema='auto', verbose=False):
+    """
+    Helper function to validate given document against a schema
+
+    Schemas are searched for locally, or within the central CMSMonitoring
+    schemas.
+    """
+    global _local_schemas
+    if schema is None:
+        return False
+
+    if _local_schemas is None:
+        # First time running, try to find the schema locally
+        # Second time, _local_schemas will be a dictionary and this is skipped
+        _local_schemas = {}
+        if os.path.isfile(schema):
+            try:
+                _local_schemas[schema] = json.load(open(schema))
+                logging.warning('Successfully loaded local schema {} for validation'.format(schema))
+            except ValueError:
+                logging.warning('Local schema {} is not json compliant'.format(schema))
+
+    if schema in _local_schemas:
+        return validate_schema(_local_schemas[schema], doc, verbose)
+
+    elif schema in _schemas.schemas():
+        return validate_schema(_schemas.schemas()[schema], doc, verbose)
+
+    else:
+        for sname, sch in _schemas.schemas().items():
+            if validate_schema(sch, doc, verbose):
+                return True
 
-def validate(data, verbose=False):
```

Comment:
Hi Valentin,
I changed that because we moved the calling of the function into `make_notification`, where we can apply it to the body directly. So there is no need to guess about a possible sub field. I think it's cleaner this way, and there is no potential for pitfalls when someone changes the name of the sub field. If some clients want to have a subfield inside their document body, they could simply adjust their schema and still run validation.